src/libdrakvuf/vmi.c

```diff
@@ -141,6 +141,17 @@ static inline void flush_vmi(drakvuf_t drakvuf)
     }
 }
 
+static void free_proc_data_priv(proc_data_priv_t* proc_data)
+{
+    g_free((gpointer)proc_data->name);
+}
+
+static void free_proc_data_priv_2(proc_data_priv_t* proc_data, proc_data_priv_t* attached_proc_data)
```

Comment:
I would prefer if there was only a single function for free proc data and just call it with NULL if there is no attached_proc_data needing to be freed

Reply:
Done

Reply:
@tklengyel I have a question. I noticed that `attached_proc_data` and `proc_data` fields in `drakvuf_trap_info_t` structure are redundant. In windows plugins always use `attached_proc_data` field and does not use `proc_data`. Linux plugins, on the contrary, always use `proc_data` field.

It would be better to leave only one fileld in the structure, `proc_data`. And fill it with attached process data on Windows if attached process was found. This would slightly optimize the hot path of drakvuf and remove the couse of common errors in plugins.

Reply:
Could you change the name of the function to be without `_2`?

Reply:
This would break implicit convention for naming of destructor functions.

`struct data_type_t { ... };
void free_data_type(struct data_type_t* d);`

Reply:
Fine